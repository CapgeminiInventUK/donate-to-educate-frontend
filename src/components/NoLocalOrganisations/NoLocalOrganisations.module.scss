@use '@assets/global.scss' as g;

.container {
  position: relative;
  color: g.$primary-dark-blue;
  display: flex;
  flex-direction: column;
}

.contactUs {
  text-decoration: underline;
}

.container svg {
  margin: 2rem 0;
}
