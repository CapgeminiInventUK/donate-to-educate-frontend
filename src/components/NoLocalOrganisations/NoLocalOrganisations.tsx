import { FC } from 'react';
import styles from './NoLocalOrganisations.module.scss';
import HandHeart from '@/assets/logo/HandHeart';
import { Link } from 'react-router-dom';
import Paths from '@/config/paths';

interface NoLocalOrganisationsProps {
  organisationName?: string;
}

const NoLocalOrganisations: FC<NoLocalOrganisationsProps> = ({
  organisationName = 'charities',
}: NoLocalOrganisationsProps) => {
  return (
    <div className={styles.container}>
      <h3>No {organisationName} have registered within your search area</h3>
      <p>
        <Link className={styles.contactUs} to={Paths.CONTACT}>
          Contact us
        </Link>
        <span> for more help.</span>
      </p>
      <div>
        <HandHeart />
      </div>
    </div>
  );
};

export default NoLocalOrganisations;
