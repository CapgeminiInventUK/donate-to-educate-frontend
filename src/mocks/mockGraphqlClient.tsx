import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { HttpResponse, graphql } from 'msw';
import { ComponentType, FC, ReactNode } from 'react';
import { server } from './server';
import { BrowserRouter as Router } from 'react-router-dom';

const createTestQueryClient = (): QueryClient =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

export const createWrapper = (node: ReactNode): ComponentType => {
  const testQueryClient = createTestQueryClient();
  const MockComponent: FC = () => {
    return (
      <Router>
        <QueryClientProvider client={testQueryClient}>{node}</QueryClientProvider>
      </Router>
    );
  };
  return MockComponent;
};

export const mockApiResponse = (
  queryName: string,
  response: Record<string, unknown> | null,
  throwError = false
): void => {
  server.use(
    graphql.query(queryName, () => {
      // eslint-disable-next-line no-console
      console.log('here');
      if (throwError) {
        return HttpResponse.json({ data: { isError: true } });
      }
      return HttpResponse.json(response);
    })
  );
};
