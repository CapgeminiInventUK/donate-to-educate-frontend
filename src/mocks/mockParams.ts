export const validPostcode = 'LL70 9DJ';
export const invalidPostcode = '123';
export const nonExistentPostcode = 'ZZ1 9ZZ';
