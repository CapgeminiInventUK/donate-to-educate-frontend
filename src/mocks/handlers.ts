import {
  AsyncResponseResolverReturnType,
  GraphQLQuery,
  HttpResponse,
  graphql,
  http,
  passthrough,
} from 'msw';
import getSchoolQueryResponse from './data/getSchoolQuery.json';
import getSchoolsNearbyQueryResponse from './data/getSchoolsNearbyQuery.json';
import { nonExistentPostcode, validPostcode } from './mockParams';

export const handlers = [
  graphql.query('GetSchool', ({ variables }) => {
    if (!String(variables.name).includes('Error')) {
      return HttpResponse.json({
        data: getSchoolQueryResponse,
      });
    }
    return HttpResponse.error() as AsyncResponseResolverReturnType<GraphQLQuery>;
  }),
  graphql.query('GetSchoolsNearby', ({ variables }) => {
    const { postcode } = variables;
    if (postcode === validPostcode) {
      return HttpResponse.json({
        data: getSchoolsNearbyQueryResponse,
      });
    }
    if (postcode === nonExistentPostcode) {
      return HttpResponse.error() as AsyncResponseResolverReturnType<GraphQLQuery>;
    }
  }),
  http.post('https://cognito-identity.eu-west-2.amazonaws.com/', () => {
    return passthrough();
  }),
  http.get('/*.webp', () => {
    return passthrough();
  }),
];
