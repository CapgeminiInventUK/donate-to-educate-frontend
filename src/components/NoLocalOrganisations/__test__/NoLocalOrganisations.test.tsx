import { createWrapper } from '@/mocks/mockGraphqlClient';
import NoLocalOrganisations from '../NoLocalOrganisations';
import { render } from '@testing-library/react';

describe('No local organisations', () => {
  it('should render component', () => {
    const Component = createWrapper(<NoLocalOrganisations />);
    expect(render(<Component />)).toMatchSnapshot();
  });
});
