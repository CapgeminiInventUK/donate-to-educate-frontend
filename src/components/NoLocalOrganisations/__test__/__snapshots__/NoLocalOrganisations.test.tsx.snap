// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`No local organisations > should render component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_317c8e"
      >
        <h3>
          No 
          charities
           have registered within your search area
        </h3>
        <p>
          <a
            class="_contactUs_317c8e"
            href="/contact"
          >
            Contact us
          </a>
          <span>
             for more help.
          </span>
        </p>
        <div>
          <svg
            class=""
            fill="none"
            height="101"
            viewBox="0 0 106 101"
            width="106"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Group 2"
            >
              <g
                filter="url(#filter0_d_30_1381)"
                id="Group 3"
              >
                <path
                  d="M9.38827 45.617L40.0553 80.3166L51.5554 91.2222C51.5554 91.2222 55.3891 95.1878 59.2221 91.2222C63.026 87.2866 59.636 83.3509 59.5845 83.2916L59.5815 83.2881L51.651 74.1724C51.2492 73.7105 51.2319 73.0285 51.6096 72.5468C52.0875 71.9373 52.9937 71.8904 53.532 72.4473L62.0971 81.308L62.0971 81.308C62.0982 81.3091 65.9306 85.2732 69.7635 81.3081C73.1724 77.7815 71.2766 74.2549 70.8149 73.5172C70.7544 73.4205 70.6824 73.3355 70.6032 73.2536L61.1063 63.4289C60.59 62.8948 60.59 62.0475 61.1063 61.5133C61.648 60.9529 62.5463 60.9529 63.088 61.5133L72.6389 71.3939C72.6389 71.3939 76.4723 75.3595 80.3052 71.3939C84.1382 67.4282 80.6674 63.4625 80.6674 63.4625L71.6063 53.5264C71.1093 52.9814 71.1215 52.1439 71.6341 51.6135C72.1834 51.0453 73.0943 51.0453 73.6437 51.6135L82.2223 60.4883C82.2223 60.4883 87.014 64.4539 89.8887 60.4883C92.7633 56.5226 89.8891 52.5569 89.8891 52.5569L82.2223 44.6256C82.989 45.4187 62.0971 23.1449 51.5554 11.9088C51.5554 11.9088 31.4298 -6.92813 13.2216 11.9088C-4.9865 30.7458 9.38827 45.617 9.38827 45.617Z"
                  fill="#E2E2E2"
                  fill-opacity="0.5"
                  id="Vector 18"
                />
                <path
                  d="M11.1568 43.5759L41.8238 78.2755L53.3239 89.1811C53.3239 89.1811 57.1577 93.1467 60.9907 89.1811C64.7946 85.2455 61.4046 81.3098 61.353 81.2505L61.35 81.247L53.4196 72.1313C53.0178 71.6694 53.0004 70.9874 53.3782 70.5057C53.8561 69.8962 54.7622 69.8493 55.3005 70.4062L63.8656 79.2669L63.8657 79.2669C63.8667 79.268 67.6992 83.2321 71.532 79.2669C74.9409 75.7404 73.0451 72.2138 72.5835 71.4761C72.523 71.3794 72.451 71.2944 72.3717 71.2125L62.8748 61.3878C62.3585 60.8536 62.3585 60.0064 62.8748 59.4722C63.4165 58.9118 64.3148 58.9118 64.8565 59.4722L74.4074 69.3528C74.4074 69.3528 78.2408 73.3184 82.0738 69.3528C85.9068 65.3871 82.436 61.4214 82.436 61.4214L73.3749 51.4853C72.8778 50.9403 72.89 50.1028 73.4027 49.5724C73.952 49.0042 74.8629 49.0042 75.4122 49.5724L83.9909 58.4472C83.9909 58.4472 88.7826 62.4128 91.6572 58.4472C94.5318 54.4815 91.6576 50.5158 91.6576 50.5158L83.9909 42.5845C84.7575 43.3776 63.8657 21.1038 53.3239 9.8677C53.3239 9.8677 33.1983 -8.96924 14.9902 9.8677C-3.21795 28.7046 11.1568 43.5759 11.1568 43.5759Z"
                  fill="rgb(136,153,182)"
                  id="Vector 16"
                />
                <g
                  filter="url(#filter1_d_30_1381)"
                  id="Rectangle 2563"
                >
                  <rect
                    fill="rgb(164,198,242)"
                    height="12.6689"
                    rx="6.33443"
                    transform="matrix(0.695012 -0.718998 0.695012 0.718998 10.374 53.3311)"
                    width="23.1948"
                  />
                </g>
                <g
                  filter="url(#filter2_d_30_1381)"
                  id="Rectangle 2564"
                >
                  <rect
                    fill="rgb(164,198,242)"
                    height="12.3956"
                    rx="6.19782"
                    transform="matrix(0.695012 -0.718998 0.695012 0.718998 19.6931 64.3107)"
                    width="28.5047"
                  />
                </g>
                <g
                  filter="url(#filter3_d_30_1381)"
                  id="Rectangle 2565"
                >
                  <rect
                    fill="rgb(164,198,242)"
                    height="12.2962"
                    rx="6.14812"
                    transform="matrix(0.695012 -0.718998 0.695012 0.718998 29.3508 74.689)"
                    width="23.9886"
                  />
                </g>
                <g
                  filter="url(#filter4_d_30_1381)"
                  id="Rectangle 2566"
                >
                  <rect
                    fill="rgb(164,198,242)"
                    height="12.5048"
                    rx="6.25239"
                    transform="matrix(0.695012 -0.718998 0.695012 0.718998 39.0618 84.7357)"
                    width="17.1292"
                  />
                </g>
                <g
                  filter="url(#filter5_i_30_1381)"
                  id="Vector 17"
                >
                  <path
                    d="M58.2704 8.90098L45.3838 29.2509C45.2936 29.3934 45.219 29.5422 45.1686 29.7032C44.8074 30.856 43.5141 35.8769 47.7443 37.6273C52.5089 39.5989 55.3783 35.6894 55.4108 35.6449L55.4126 35.6424L64.4324 23.145C65.3211 21.9137 67.1108 21.795 68.1543 22.8981L89.8239 45.8044C90.6441 46.6714 91.9651 46.801 92.848 45.998C94.8051 44.2182 98.0795 40.6487 99.8736 35.4138C102.908 26.5598 101.132 12.7562 92.9778 5.90199C85.9779 0.0182489 77.0087 -0.916718 70.165 0.714144C62.3062 2.5869 58.5561 8.44069 58.3002 8.85326L58.295 8.86164C58.2867 8.87492 58.2791 8.88731 58.2704 8.90098Z"
                    fill="rgb(164,198,242)"
                  />
                </g>
              </g>
            </g>
            <defs>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="100.985"
                id="filter0_d_30_1381"
                width="105.264"
                x="0"
                y="0"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="4"
                />
                <fegaussianblur
                  stdDeviation="2"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.04 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_30_1381"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_30_1381"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="28.45"
                id="filter1_d_30_1381"
                width="27.7678"
                x="8.95288"
                y="39.322"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="4"
                />
                <fegaussianblur
                  stdDeviation="2"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.04 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_30_1381"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_30_1381"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="32.1865"
                id="filter2_d_30_1381"
                width="31.3796"
                x="18.2163"
                y="46.4263"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="4"
                />
                <fegaussianblur
                  stdDeviation="2"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.04 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_30_1381"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_30_1381"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="28.9098"
                id="filter3_d_30_1381"
                width="28.2122"
                x="27.854"
                y="60.0307"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="4"
                />
                <fegaussianblur
                  stdDeviation="2"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.04 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_30_1381"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_30_1381"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="24.04"
                id="filter4_d_30_1381"
                width="23.5051"
                x="37.6072"
                y="75.0532"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="4"
                />
                <fegaussianblur
                  stdDeviation="2"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.04 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="effect1_dropShadow_30_1381"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_30_1381"
                  mode="normal"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="50.5333"
                id="filter5_i_30_1381"
                width="56.5652"
                x="44.6985"
                y="0"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <feblend
                  in="SourceGraphic"
                  in2="BackgroundImageFix"
                  mode="normal"
                  result="shape"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset
                  dy="4"
                />
                <fegaussianblur
                  stdDeviation="5"
                />
                <fecomposite
                  in2="hardAlpha"
                  k2="-1"
                  k3="1"
                  operator="arithmetic"
                />
                <fecolormatrix
                  type="matrix"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.1 0"
                />
                <feblend
                  in2="shape"
                  mode="normal"
                  result="effect1_innerShadow_30_1381"
                />
              </filter>
            </defs>
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_317c8e"
    >
      <h3>
        No 
        charities
         have registered within your search area
      </h3>
      <p>
        <a
          class="_contactUs_317c8e"
          href="/contact"
        >
          Contact us
        </a>
        <span>
           for more help.
        </span>
      </p>
      <div>
        <svg
          class=""
          fill="none"
          height="101"
          viewBox="0 0 106 101"
          width="106"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="Group 2"
          >
            <g
              filter="url(#filter0_d_30_1381)"
              id="Group 3"
            >
              <path
                d="M9.38827 45.617L40.0553 80.3166L51.5554 91.2222C51.5554 91.2222 55.3891 95.1878 59.2221 91.2222C63.026 87.2866 59.636 83.3509 59.5845 83.2916L59.5815 83.2881L51.651 74.1724C51.2492 73.7105 51.2319 73.0285 51.6096 72.5468C52.0875 71.9373 52.9937 71.8904 53.532 72.4473L62.0971 81.308L62.0971 81.308C62.0982 81.3091 65.9306 85.2732 69.7635 81.3081C73.1724 77.7815 71.2766 74.2549 70.8149 73.5172C70.7544 73.4205 70.6824 73.3355 70.6032 73.2536L61.1063 63.4289C60.59 62.8948 60.59 62.0475 61.1063 61.5133C61.648 60.9529 62.5463 60.9529 63.088 61.5133L72.6389 71.3939C72.6389 71.3939 76.4723 75.3595 80.3052 71.3939C84.1382 67.4282 80.6674 63.4625 80.6674 63.4625L71.6063 53.5264C71.1093 52.9814 71.1215 52.1439 71.6341 51.6135C72.1834 51.0453 73.0943 51.0453 73.6437 51.6135L82.2223 60.4883C82.2223 60.4883 87.014 64.4539 89.8887 60.4883C92.7633 56.5226 89.8891 52.5569 89.8891 52.5569L82.2223 44.6256C82.989 45.4187 62.0971 23.1449 51.5554 11.9088C51.5554 11.9088 31.4298 -6.92813 13.2216 11.9088C-4.9865 30.7458 9.38827 45.617 9.38827 45.617Z"
                fill="#E2E2E2"
                fill-opacity="0.5"
                id="Vector 18"
              />
              <path
                d="M11.1568 43.5759L41.8238 78.2755L53.3239 89.1811C53.3239 89.1811 57.1577 93.1467 60.9907 89.1811C64.7946 85.2455 61.4046 81.3098 61.353 81.2505L61.35 81.247L53.4196 72.1313C53.0178 71.6694 53.0004 70.9874 53.3782 70.5057C53.8561 69.8962 54.7622 69.8493 55.3005 70.4062L63.8656 79.2669L63.8657 79.2669C63.8667 79.268 67.6992 83.2321 71.532 79.2669C74.9409 75.7404 73.0451 72.2138 72.5835 71.4761C72.523 71.3794 72.451 71.2944 72.3717 71.2125L62.8748 61.3878C62.3585 60.8536 62.3585 60.0064 62.8748 59.4722C63.4165 58.9118 64.3148 58.9118 64.8565 59.4722L74.4074 69.3528C74.4074 69.3528 78.2408 73.3184 82.0738 69.3528C85.9068 65.3871 82.436 61.4214 82.436 61.4214L73.3749 51.4853C72.8778 50.9403 72.89 50.1028 73.4027 49.5724C73.952 49.0042 74.8629 49.0042 75.4122 49.5724L83.9909 58.4472C83.9909 58.4472 88.7826 62.4128 91.6572 58.4472C94.5318 54.4815 91.6576 50.5158 91.6576 50.5158L83.9909 42.5845C84.7575 43.3776 63.8657 21.1038 53.3239 9.8677C53.3239 9.8677 33.1983 -8.96924 14.9902 9.8677C-3.21795 28.7046 11.1568 43.5759 11.1568 43.5759Z"
                fill="rgb(136,153,182)"
                id="Vector 16"
              />
              <g
                filter="url(#filter1_d_30_1381)"
                id="Rectangle 2563"
              >
                <rect
                  fill="rgb(164,198,242)"
                  height="12.6689"
                  rx="6.33443"
                  transform="matrix(0.695012 -0.718998 0.695012 0.718998 10.374 53.3311)"
                  width="23.1948"
                />
              </g>
              <g
                filter="url(#filter2_d_30_1381)"
                id="Rectangle 2564"
              >
                <rect
                  fill="rgb(164,198,242)"
                  height="12.3956"
                  rx="6.19782"
                  transform="matrix(0.695012 -0.718998 0.695012 0.718998 19.6931 64.3107)"
                  width="28.5047"
                />
              </g>
              <g
                filter="url(#filter3_d_30_1381)"
                id="Rectangle 2565"
              >
                <rect
                  fill="rgb(164,198,242)"
                  height="12.2962"
                  rx="6.14812"
                  transform="matrix(0.695012 -0.718998 0.695012 0.718998 29.3508 74.689)"
                  width="23.9886"
                />
              </g>
              <g
                filter="url(#filter4_d_30_1381)"
                id="Rectangle 2566"
              >
                <rect
                  fill="rgb(164,198,242)"
                  height="12.5048"
                  rx="6.25239"
                  transform="matrix(0.695012 -0.718998 0.695012 0.718998 39.0618 84.7357)"
                  width="17.1292"
                />
              </g>
              <g
                filter="url(#filter5_i_30_1381)"
                id="Vector 17"
              >
                <path
                  d="M58.2704 8.90098L45.3838 29.2509C45.2936 29.3934 45.219 29.5422 45.1686 29.7032C44.8074 30.856 43.5141 35.8769 47.7443 37.6273C52.5089 39.5989 55.3783 35.6894 55.4108 35.6449L55.4126 35.6424L64.4324 23.145C65.3211 21.9137 67.1108 21.795 68.1543 22.8981L89.8239 45.8044C90.6441 46.6714 91.9651 46.801 92.848 45.998C94.8051 44.2182 98.0795 40.6487 99.8736 35.4138C102.908 26.5598 101.132 12.7562 92.9778 5.90199C85.9779 0.0182489 77.0087 -0.916718 70.165 0.714144C62.3062 2.5869 58.5561 8.44069 58.3002 8.85326L58.295 8.86164C58.2867 8.87492 58.2791 8.88731 58.2704 8.90098Z"
                  fill="rgb(164,198,242)"
                />
              </g>
            </g>
          </g>
          <defs>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="100.985"
              id="filter0_d_30_1381"
              width="105.264"
              x="0"
              y="0"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecomposite
                in2="hardAlpha"
                operator="out"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.04 0"
              />
              <feblend
                in2="BackgroundImageFix"
                mode="normal"
                result="effect1_dropShadow_30_1381"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_dropShadow_30_1381"
                mode="normal"
                result="shape"
              />
            </filter>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="28.45"
              id="filter1_d_30_1381"
              width="27.7678"
              x="8.95288"
              y="39.322"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecomposite
                in2="hardAlpha"
                operator="out"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.04 0"
              />
              <feblend
                in2="BackgroundImageFix"
                mode="normal"
                result="effect1_dropShadow_30_1381"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_dropShadow_30_1381"
                mode="normal"
                result="shape"
              />
            </filter>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="32.1865"
              id="filter2_d_30_1381"
              width="31.3796"
              x="18.2163"
              y="46.4263"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecomposite
                in2="hardAlpha"
                operator="out"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.04 0"
              />
              <feblend
                in2="BackgroundImageFix"
                mode="normal"
                result="effect1_dropShadow_30_1381"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_dropShadow_30_1381"
                mode="normal"
                result="shape"
              />
            </filter>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="28.9098"
              id="filter3_d_30_1381"
              width="28.2122"
              x="27.854"
              y="60.0307"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecomposite
                in2="hardAlpha"
                operator="out"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.04 0"
              />
              <feblend
                in2="BackgroundImageFix"
                mode="normal"
                result="effect1_dropShadow_30_1381"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_dropShadow_30_1381"
                mode="normal"
                result="shape"
              />
            </filter>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="24.04"
              id="filter4_d_30_1381"
              width="23.5051"
              x="37.6072"
              y="75.0532"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="2"
              />
              <fecomposite
                in2="hardAlpha"
                operator="out"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.04 0"
              />
              <feblend
                in2="BackgroundImageFix"
                mode="normal"
                result="effect1_dropShadow_30_1381"
              />
              <feblend
                in="SourceGraphic"
                in2="effect1_dropShadow_30_1381"
                mode="normal"
                result="shape"
              />
            </filter>
            <filter
              color-interpolation-filters="sRGB"
              filterUnits="userSpaceOnUse"
              height="50.5333"
              id="filter5_i_30_1381"
              width="56.5652"
              x="44.6985"
              y="0"
            >
              <feflood
                flood-opacity="0"
                result="BackgroundImageFix"
              />
              <feblend
                in="SourceGraphic"
                in2="BackgroundImageFix"
                mode="normal"
                result="shape"
              />
              <fecolormatrix
                in="SourceAlpha"
                result="hardAlpha"
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
              />
              <feoffset
                dy="4"
              />
              <fegaussianblur
                stdDeviation="5"
              />
              <fecomposite
                in2="hardAlpha"
                k2="-1"
                k3="1"
                operator="arithmetic"
              />
              <fecolormatrix
                type="matrix"
                values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.1 0"
              />
              <feblend
                in2="shape"
                mode="normal"
                result="effect1_innerShadow_30_1381"
              />
            </filter>
          </defs>
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
